apps/hash-frontend/src/pages/drafts.page/draft-entity/draft-entity-web.tsx

```diff
@@ -0,0 +1,77 @@
+import { Entity, extractOwnedByIdFromEntityId } from "@local/hash-subgraph";
+import { FunctionComponent, useMemo } from "react";
+
+import { useOrgs } from "../../../components/hooks/use-orgs";
+import { useUsers } from "../../../components/hooks/use-users";
+import { UserIcon } from "../../../shared/icons/user-icon";
+import { UsersRegularIcon } from "../../../shared/icons/users-regular-icon";
+import { Link } from "../../../shared/ui";
+import { useAuthenticatedUser } from "../../shared/auth-info-context";
+import { DraftEntityChip } from "./draft-entity-chip";
+
+export const DraftEntityWeb: FunctionComponent<{ entity: Entity }> = ({
+  entity,
+}) => {
+  const ownedById = useMemo(
+    () => extractOwnedByIdFromEntityId(entity.metadata.recordId.entityId),
+    [entity],
+  );
+
+  const { authenticatedUser } = useAuthenticatedUser();
+  const { orgs } = useOrgs();
+  const { users } = useUsers();
+
+  const web = useMemo(() => {
+    if (!orgs || !users) {
+      return undefined;
+    }
+
+    const org = orgs.find(({ accountGroupId }) => accountGroupId === ownedById);
+
+    if (org) {
+      return org;
+    }
+
+    if (authenticatedUser.accountId === ownedById) {
+      return authenticatedUser;
+    }
+
+    const user = users.find(({ accountId }) => accountId === ownedById);
+
+    if (user) {
+      return user;
+    }
+
+    throw new Error(
+      `Could not find web of draft entity with ownedById ${ownedById}`,
+    );
+  }, [ownedById, orgs, users, authenticatedUser]);
+
+  const label = useMemo(() => {
+    if (!web) {
+      return undefined;
+    }
+
+    return web.kind === "user"
+      ? web.accountId === authenticatedUser.accountId
+        ? "My Web"
+        : web.preferredName
+          ? web.preferredName
+          : "Unknown User"
+      : web.name;
+  }, [web, authenticatedUser]);
```

Comment:
@nonparibus in [the ticket](https://linear.app/hash/issue/H-1550/drafts-page-replace-visible-to-x-people-with-name-of-containing-web-eg) for replacing the "Visible to X people" indicator with name of containing web it says to use the shortname to identify the web, but I've instead used the name of the user or the web to be consistent with the values in the "Web" filter section on the right hand side because I thought this made more sense. I've also made this a clickable chip. Let me know if you have any thoughts on this, and would prefer to use the shortname here.

https://github.com/hashintel/hash/assets/42802102/5700dfdd-f5fc-4c8d-8352-212fe4236653